Content.Server/GameObjects/Components/Interactable/MatchstickComponent.cs

```diff
@@ -0,0 +1,112 @@
+﻿#nullable enable
+using System.Threading.Tasks;
+using Content.Shared.Audio;
+using Content.Shared.GameObjects.Components;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Server.GameObjects;
+using Robust.Server.GameObjects.EntitySystems;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Components.Timers;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Interactable
+{
+    [RegisterComponent]
+    [ComponentReference(typeof(IHotItem))]
+    public class MatchstickComponent : Component, IHotItem, IInteractUsing
+    {
+        public override string Name => "Matchstick";
+
+        private SharedBurningStates _currentState;
```

Comment:
can just be initialized here instead of in intialize
```suggestion
        private SharedBurningStates _currentState = SharedBurningStates.Unlit;
```